SwiftWisdom/Core/StandardLibrary/Numbers/Mathable.swift

```diff
@@ -0,0 +1,73 @@
+//
+//  Mathable.swift
+//  SwiftWisdom
+//
+//  Created by Paul Rolfe on 2/12/16.
+//  Copyright © 2016 Intrepid. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+public protocol Mathable: Comparable {
+    func + (lhs: Self, rhs: Self) -> Self
+    func / (lhs: Self, rhs: Self) -> Self
+    func * (lhs: Self, rhs: Self) -> Self
+    func - (lhs: Self, rhs: Self) -> Self
+    init(_ int: IntMax)
+}
+
+extension Mathable {
+    public static var zero: Self {
+        return self.init(0.toIntMax())
```

Comment:
Yea, the reason we use `IntMax` is because almost all primitive numeric types can take it as an initializer.  My guess is that under the covers, this is how things like this work:

```
let a: Double = 0
```

etc.